src/Checkbox/Checkbox.tsx

```diff
@@ -0,0 +1,24 @@
+import React, { useEffect, useRef } from 'react';
+import { StyledCheckbox } from './Checkbox.styles';
+import { CheckboxProps } from './types';
+
+export const Checkbox = ({ checked, disabled, indeterminate, onChange, ...rest }: CheckboxProps) => {
+  const checkboxRef = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    if (checkboxRef && checkboxRef.current) {
+      checkboxRef.current.indeterminate = Boolean(indeterminate);
```

Comment:
If we already tell to typescript that indeterminate is boolean we need to do `Boolean(indeterminate)` ?

Reply:
Yes, I think we should not set indeterminate DOM property as undefined

Reply:
Aaaaandaa.
And if we give a default value to indeterminate?

Reply:
Ok, changing it